src/systems/imu/Imu.cc

```diff
@@ -194,6 +195,18 @@ void ImuPrivate::addIMU(
         data.ImuSensor()->OrientationEnabled());
   }
 
+  // Get world frame orientation and heading
+  double heading = 0;
+
+  ignition::gazebo::World world(worldEntity);
+  if (world.SphericalCoordinates(_ecm)) {
+    auto sphericalCoordinates = world.SphericalCoordinates(_ecm).value();
+    heading = sphericalCoordinates.HeadingOffset().Radian();
+  }
+
+  sensor->SetWorldFrameOrientation(math::Quaterniond(0, 0, heading),
```

Comment:
Should this function only be executed if the `World` has `SphericalCoordinates`?

Reply:
``sensor->SetWorldFrameOrientation`` should be run irrespective of whether the World has ``SphericalCoordinates`` because that is where the ``<localization>`` tag in IMU is parsed. Added here : https://github.com/ignitionrobotics/ign-sensors/blob/33a79c6524e69dbf4006815458cddef1fec3f785/src/ImuSensor.cc#L315

For the ``heading = ....`` line, that was just to make sure it defaults to 0,  and is then overridden if the world has spherical coordinates. Maybe I can remove that if ``sphericalCoordinates`` has a default with 0 heading.